### PR TITLE
Resolve Auth for TypeScript Execution Test

### PR DESCRIPTION
**Pull request by rkodev**
1. Provides an authentication token for the TS Execution Tests
2. Improve error string serialization for execution tests
3. Ignore snippets failing compilation for TS execution

### CODE REVIEW

File: TestsCommon/TypeScriptTestRunner.cs [hunk 1]
```diff
@@ -1,78 +1,149 @@
...
+                // warm up application tokens since only 2 distinct token posibilities exist
+                var warmUpTasks = new List<Task> {
+                    GetApplicationAccessToken(false),
+                    GetApplicationAccessToken(true),
+                };
+                await Task.WhenAll(warmUpTasks).ConfigureAwait(false);
+            }
+        }
+
+        private async Task generateRequestInformationFiles(IEnumerable<LanguageTestData> languageTestData)
```
Comment (rkodev) [inline TestsCommon/TypeScriptTestRunner.cs:115]: To extract the http url, needed later to get the scopes available, this section extracts the snippet and transforms it from 

```
const graphServiceClient = msgraph_sdk_javascript_1.GraphServiceClient.init({ authProvider });
const result = async () => {
    await graphServiceClient.identityGovernance.entitlementManagement.accessPackageAssignmentsById("accessPackageAssignment-id").get();
};
result().then(resp => {
    console.log(resp);
});

```

to 

```
const graphServiceClient = msgraph_sdk_javascript_1.GraphServiceClient.init({ authProvider });
const req =  graphServiceClient.identityGovernance.entitlementManagement.accessPackageAssignmentsById("accessPackageAssignment-id").createGetRequestInformation()
const url = req.URL
const method = req.httpMethod;
console.log("url: " + url);
console.log("httpMethod: " + method);

```

File: TestsCommon/TypeScriptTestRunner.cs [hunk 2]
```diff
@@ -144,7 +215,8 @@ private async static Task dumpFiles(string compilationDirectory, IEnumerable<Lan
                 var codeToCompile = SDKShellTemplate
                                         .Replace("//insert-code-here", codeSnippetFormatted)
                                         .Replace("\r\n", "\n").Replace("\n", "\r\n")
-                                        .Replace("--imports--", generatedImports);
+                                        .Replace("//insert-imports-here", generatedImports)
```
Comment (rkodev) [inline TestsCommon/TypeScriptTestRunner.cs:218]: Executes the promise and by appending the suffix below 
```
result().then(resp => {
    console.log(resp);
});
```

File: TestsCommon/TypeScriptTestRunner.cs [hunk 3]
```diff
@@ -264,76 +329,221 @@ public async Task RunCompilationTests(LanguageTestData testData)
...
         /// <summary>
-        /// 1. Fetches snippet from docs repo
-        /// 2. Asserts that there is one and only one snippet in the file
-        /// 3. Wraps snippet with compilable template
-        /// 4. Attempts to compile and reports errors if there is any
+        /// Returns a pair holding (url , httpMethod) extracted from a typecsript call
         /// </summary>
-        /// <param name="testData">Test data containing information such as snippet file name</param>
-        public async Task RunExecutionTests(LanguageTestData testData)
+        /// <returns></returns>
```
Comment (rkodev) [inline TestsCommon/TypeScriptTestRunner.cs:341]: Lines 341-524 -> Executes the js files by first attempting only Application Access before falling back to delegated access. 
NB: Delegated access is significantly slower. About 220 of the 570+ rely on delegated access

**Comment by rkodev** [comment]
@zengin Without delegated access the pipeline runs to completion with  a [48% pass rate ](https://microsoftgraph.visualstudio.com/Graph%20Developer%20Experiences/_build/results?buildId=79243&view=ms.vss-test-web.build-test-results-tab)

With delegated access enable the pipeline is [not able to execute to completion](https://microsoftgraph.visualstudio.com/Graph%20Developer%20Experiences/_build/results?buildId=79374&view=results)

File: azure-pipelines/typescript-v1-execution-tests.yml [hunk 4]
```diff
@@ -34,7 +34,7 @@ variables:
   buildConfiguration: 'Release'
 
 pool:
-  name: 1es-ubuntu-latest
+  name: 1es-ubuntu-msgraph-sdk-raptor-compute-latest
```
Comment (rkodev) [inline azure-pipelines/typescript-v1-execution-tests.yml:37]: Requires more resource , [please check](https://microsoftgraph.visualstudio.com/Graph%20Developer%20Experiences/_build/results?buildId=79374&view=results)

Compiling all the tests and elimination of all compilation errors managed to drop execution from 2-3 minutes to about 1-20 seconds.  However this looks like the only option left

File: TestsCommon/TypeScriptTestRunner.cs [hunk 5]
```diff
@@ -264,76 +329,221 @@ public async Task RunCompilationTests(LanguageTestData testData)
...
+            if (currentToken == false && string.IsNullOrWhiteSpace(tokenValue))
+            {
+                var permissionManager = await GetPermissionManager(isEducation).ConfigureAwait(false);
+                var tokenContext = new TokenRequestContext(new[] { ResourceConstants.DefaultAuthScope });
+                try
+                {
+                    var token = await permissionManager.ClientCertificateCredential.GetTokenAsync(tokenContext)
+                        .ConfigureAwait(false);
+                    tokenValue = token.Token;
+                    ApplicationAccessTokenCache[isEducation] = tokenValue;
```
Comment (zengin) [inline TestsCommon/TypeScriptTestRunner.cs:380]: Do we have auth libraries for TS? It might be better to use those to have end-to-end coverage.

**Review by zengin — CHANGES_REQUESTED** [review]

File: TestsCommon/GoTestRunner.cs [hunk 6]
```diff
@@ -151,16 +151,6 @@ private static string ReplaceHyphensWithUnderscores(string stringContent)
         return stringContent.Replace("-", "_");
     }
 
-     /// <summary>
```
Comment (fey101) [inline TestsCommon/GoTestRunner.cs:154]: Any reason for removing this?

Comment (fey101) [reply TestsCommon/GoTestRunner.cs:154]: No worries, seen it's a refactor into TestsCommon.stringsExtensions

**Pull request merged**